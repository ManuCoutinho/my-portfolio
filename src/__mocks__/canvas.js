HTMLCanvasElement.prototype.getContext = () => {
  return {
    fillRect: jest.fn()
  }
}
