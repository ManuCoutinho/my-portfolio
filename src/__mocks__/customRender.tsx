import { ReactElement, FC, ReactNode } from 'react'
import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
	return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
