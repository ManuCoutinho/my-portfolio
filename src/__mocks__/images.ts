export const mockImg = [
	{
		imgPng: '/img1.png',
		imgWebp: '/img1.webp',
		alt: 'img1',
		id: 'img1'
	},
	{
		imgPng: '/img2.png',
		imgWebp: '/img2.webp',
		alt: 'img2',
		id: 'img2'
	}
]
