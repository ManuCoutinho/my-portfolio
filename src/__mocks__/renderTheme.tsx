import { ReactElement, ReactNode, FC } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
