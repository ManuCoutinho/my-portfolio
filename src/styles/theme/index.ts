import { DefaultTheme, CustomTheme } from 'styled-components'
import dark from './dark'
import light from './light'

export type Theme = typeof defaultTheme

const defaultTheme = {
	font: {
		cursive:
			"'Montserrat Alternates', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, 'Opens Sans', 'Helvetica Neue', sans-serif",
		regular:
			"'Josefin Sans', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, 'Opens Sans', 'Helvetica Neue', sans-serif"
	},
	fontSize: {
		xsm: '0.5rem', //8px
		sm: '0.875rem', //14px
		md: '1rem', //16px
		wd: '1.25rem', //20px
		lg: '1.5rem', //24px
		xl: '2rem', //36px
		xlg: '3rem' //48px
	},
	fontWeight: {
		thin: 100,
		light: 200,
		regular: 400,
		semibold: 600,
		bold: 700
	},
	radius: {
		sm: '8px',
		md: '16px',
		pill: '25px',
		full: '50%'
	}
} as const

function combineTheme(theme: CustomTheme): DefaultTheme {
	return { ...defaultTheme, ...theme }
}
export { combineTheme, dark, light }
