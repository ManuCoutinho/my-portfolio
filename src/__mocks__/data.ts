import { mockImg } from './images'
export default {
  description: 'a beutiful test',
  site: 'http://www.somesit.com',
  repo: 'http://www.somerepo.com',
  styles: 'styles',
  api: 'mock',
  framework: 'next',
  name: 'project',
  tools: ['tools'],
  id: 'id',
  doc: 'http://www.storybook.com',
  coverDefaultLight: 'img.png',
  coverDefaultDark: '',
  coverAltLight: '',
  coverAltDark: '',
  alt: 'mock img',
  img: mockImg,
  route: '/test'
}
