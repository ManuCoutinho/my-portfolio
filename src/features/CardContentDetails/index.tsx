import Link from 'next/link'
import { Icon } from '@iconify-icon/react'
import { StackBox } from 'components/StackBox'
import { Text } from 'components/Text'
import { DataProps } from 'features/Card/types'
import * as Styled from './styles'
import icons from 'constants/icons'

const CardContentDetails: React.FC<DataProps> = ({ ...props }) => {
  return (
    <StackBox id='details' direction='column' gap={1} px='1.5rem'>
      <Styled.BoxDetails>
        <Styled.Topic>Descrição: </Styled.Topic>
        <Text as='dd'>{props.description}</Text>
      </Styled.BoxDetails>
      {props.framework && (
        <Styled.BoxDetails>
          <Styled.Topic>Framework: </Styled.Topic>
          <Text as='dd'>{props.framework}</Text>
        </Styled.BoxDetails>
      )}
      <Styled.BoxDetails>
        <Styled.Topic>Linguagens e Ferramentas: </Styled.Topic>
        <Text as='dd'>{props.tools?.join(' / ')}</Text>
      </Styled.BoxDetails>
      <Styled.BoxDetails>
        <Styled.Topic>Estilização: </Styled.Topic>
        <Text as='dd'>{props.styles}</Text>
      </Styled.BoxDetails>
      {props.api && (
        <Styled.BoxDetails>
          <Styled.Topic>API: </Styled.Topic>
          <Text as='dd'>{props.api}</Text>
        </Styled.BoxDetails>
      )}
      <Styled.BoxDetails as='dd'>
        <Styled.Topic>Disponível em: </Styled.Topic>
        <Link href={props.repo} passHref>
          <Styled.IconLink
            target='_blank'
            rel='noopener noreferrer'
            title='repositorio no github'
          >
            <Icon icon={icons.github} style={{ fontSize: '1.8rem' }} />
          </Styled.IconLink>
        </Link>
        {props.site != 'Em breve' && (
          <Link href={props.site} passHref>
            <Styled.IconLink
              target='_blank'
              rel='noopener noreferrer'
              title={`acessar o ${props.name}`}
            >
              <Icon icon={icons.link_external} />
            </Styled.IconLink>
          </Link>
        )}
        {props.doc && (
          <Link href={props.doc} passHref>
            <Styled.IconLink
              target='_blank'
              rel='noopener noreferrer'
              title={`documentation for ${props.name}`}
            >
              <Icon icon={icons.docs} />
            </Styled.IconLink>
          </Link>
        )}
      </Styled.BoxDetails>
    </StackBox>
  )
}

export default CardContentDetails
