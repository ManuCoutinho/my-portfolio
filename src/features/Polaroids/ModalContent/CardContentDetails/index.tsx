import Link from 'next/link'
import { Icon } from '@iconify-icon/react'

import { StackBox } from 'components/StackBox'
import { Text } from 'components/Text'

import icons from 'constants/icons'
import { DataType } from '@models/data'
import * as Styled from './styles'

const CardContentDetails: React.FC<DataType> = ({ ...props }) => {
	return (
		<StackBox id='details' direction='column' gap={1} px='1.5rem' as='dl'>
			<Styled.BoxDetails>
				<Styled.Topic aria-label='description'>Descrição: </Styled.Topic>
				<Text as='dd' aria-label={props.description}>
					{props.description}
				</Text>
			</Styled.BoxDetails>
			{props.framework && (
				<Styled.BoxDetails>
					<Styled.Topic aria-label={'Framework'}>Framework: </Styled.Topic>
					<Text as='dd' aria-label={props.framework}>
						{props.framework}
					</Text>
				</Styled.BoxDetails>
			)}
			<Styled.BoxDetails>
				<Styled.Topic aria-label={'tools'}>
					Linguagens e Ferramentas:{' '}
				</Styled.Topic>
				<Text as='dd' aria-label={props.tools}>
					{props.tools?.join(' / ')}
				</Text>
			</Styled.BoxDetails>
			<Styled.BoxDetails>
				<Styled.Topic aria-label={'style'}>Estilização: </Styled.Topic>
				<Text as='dd' aria-label={props.styles}>
					{props.styles}
				</Text>
			</Styled.BoxDetails>
			{props.api && (
				<Styled.BoxDetails>
					<Styled.Topic aria-label={'api'}>API: </Styled.Topic>
					<Text as='dd' aria-label={props.api}>
						{props.api}
					</Text>
				</Styled.BoxDetails>
			)}
			<Styled.BoxDetails>
				<Styled.Topic aria-label={'available'}>Disponível em: </Styled.Topic>
				<Link href={props.repo} passHref>
					<Styled.IconLink
						target='_blank'
						rel='noopener noreferrer'
						title='repositorio no github'
					>
						<Icon icon={icons.github} style={{ fontSize: '1.8rem' }} />
					</Styled.IconLink>
				</Link>
				{props.site != 'Em breve' && (
					<Link href={props.site} passHref>
						<Styled.IconLink
							target='_blank'
							rel='noopener noreferrer'
							title={`acessar o ${props.name}`}
						>
							<Icon icon={icons.link_external} />
						</Styled.IconLink>
					</Link>
				)}
				{props.doc && (
					<Link href={props.doc} passHref>
						<Styled.IconLink
							target='_blank'
							rel='noopener noreferrer'
							title={`documentation for ${props.name}`}
						>
							<Icon icon={icons.docs} />
						</Styled.IconLink>
					</Link>
				)}
			</Styled.BoxDetails>
		</StackBox>
	)
}

export default CardContentDetails
