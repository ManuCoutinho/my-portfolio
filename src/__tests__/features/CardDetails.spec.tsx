import { screen } from '@testing-library/react'
import { render } from '../../styles/theme/renderTheme'
import { act } from 'react-dom/test-utils'
import CardContentDetails from '../../features/CardContentDetails'
import * as data from '../../__mocks__/data'

const mockData = data.default
describe('<CardContentDetails/>', () => {
  it('should render the links correctly', () => {
    act(async () => {
      render(<CardContentDetails {...mockData} />)
      const ghLink = await screen.findByRole('link', {
        name: /repositorio no github/i
      })
      const projectLink = await screen.findByRole('link', {
        name: /acessar o project/i
      })
      expect(ghLink).toBeInTheDocument()
      expect(ghLink).toHaveAttribute('href', mockData.repo)
      expect(projectLink).toHaveAttribute('href', mockData.site)
      expect(projectLink).toHaveAttribute('target', '_blank')
    })
  })
  it('should render the subtitles correctly', () => {
    act(async () => {
      render(<CardContentDetails {...mockData} />)
      expect(await screen.findByText(/descrição/i)).toBeInTheDocument()
      expect(await screen.findByText(/framework/i)).toBeInTheDocument()
      expect(
        await screen.findByText(/linguagens e ferramentas/i)
      ).toBeInTheDocument()
      expect(await screen.findByText(/estilização/i)).toBeInTheDocument()
      expect(await screen.findByText(/api/i)).toBeInTheDocument()
      expect(await screen.findByText(/disponível em/i)).toBeInTheDocument()
    })
  })
  it('should render a Card Details correctly', () => {
    const { container } = render(<CardContentDetails {...mockData} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
