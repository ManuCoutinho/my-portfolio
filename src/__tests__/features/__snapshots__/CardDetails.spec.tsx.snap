// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardContentDetails/> should render a Card Details correctly 1`] = `null`;
